examples/tensorflow2/scripts/tf_keras_gradienttape.py

```diff
@@ -0,0 +1,93 @@
+"""
+This script is a ResNet training script which uses Tensorflow's Keras interface.
+It has been orchestrated with SageMaker Debugger hook to allow saving tensors during training.
+Here, the hook has been created using its constructor to allow running this locally for your experimentation.
+When you want to run this script in SageMaker, it is recommended to create the hook from json file.
+"""
+
+# Standard Library
+import argparse
+
+# Third Party
+import numpy as np
+import tensorflow.compat.v2 as tf
+from tensorflow.keras.applications.resnet50 import ResNet50
+from tensorflow.keras.datasets import cifar10
+from tensorflow.keras.utils import to_categorical
+
+# First Party
+import smdebug.tensorflow as smd
+
+
+def train(batch_size, n_epochs, model, hook):
+    (x_train, y_train), _ = cifar10.load_data()
+
+    x_train = x_train.astype("float32")
+
+    mean_image = np.mean(x_train, axis=(0, 1, 2))
+    stddev_image = np.std(x_train, axis=(0, 1, 2))
+    x_train -= mean_image
+    x_train /= stddev_image
+
+    dataset = tf.data.Dataset.from_tensor_slices(
+        (tf.cast(x_train, tf.float32), tf.cast(y_train, tf.int64))
+    )
+    dataset = dataset.shuffle(1000).batch(batch_size)
+
+    optimizer = tf.keras.optimizers.Adam()
+    cce = tf.keras.losses.CategoricalCrossentropy(from_logits=True)
+    train_acc_metric = tf.keras.metrics.SparseCategoricalAccuracy()
+
+    num_training_samples = x_train.shape[0]
+
+    for epoch in range(n_epochs):
+        print("Epoch %d/%d" % (epoch + 1, n_epochs))
+        progBar = tf.keras.utils.Progbar(num_training_samples, stateful_metrics="accuracy")
+        for idx, (data, labels) in enumerate(dataset):
+            dataset_labels = labels
+            labels = to_categorical(labels, 10)
+            # wrap the tape so the hook can identify the gradients, parameters, loss
+            # wrapping the tape ensures that smdebug's wrapper around functions of
+            # the tape object - push_tape(), pop_tape(), gradient(), will setup the writers of
+            # the debugger and save tensors that are provided as input to gradient() (trainable
+            # variables and loss), output of gradient() (gradients).
+            with hook.wrap_tape(tf.GradientTape()) as tape:
```

Comment:
Ideally we can do a ZCC for this? by modifying GradientTape function correct?

Reply:
we cant do ZCC. ZCC doesn't support GradientTape yet.